[PATCH] runltp: allow symlinks to runltp

Allow symlinking of runltp script to locations outside of the ltp base
directory (such as to /usr/bin).

Signed-off-by: Jiri Benc <jbenc@suse.cz>

---
 runltp |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

Index: ltp-full-20100831/runltp
===================================================================
--- ltp-full-20100831.orig/runltp
+++ ltp-full-20100831/runltp
@@ -71,12 +71,14 @@
 
 setup()
 {
-    cd `dirname $0` || \
+    LTPROOT=$(dirname $0)
+    [ -h "$0" ] && LTPROOT=$(dirname $(readlink -m $0))
+    cd "$LTPROOT" || \
     {
-        echo "FATAL: unable to change directory to $(dirname $0)"
+        echo "FATAL: unable to change directory to $LTPROOT"
         exit 1
     }
-    export LTPROOT=${PWD}
+    export LTPROOT
     export TMPBASE="/tmp"
     export PATH="${PATH}:${LTPROOT}/testcases/bin"
 
